src/Cache.cpp

```diff
@@ -349,18 +349,38 @@ secretName(std::string name, bool internal)
 void
 Cache::loadSecrets(std::vector<std::pair<std::string, bool>> toLoad)
 {
+
+    auto settings = UserSettings::instance()->qsettings();
+
     if (toLoad.empty()) {
         this->databaseReady_ = true;
         emit databaseReady();
+        nhlog::db()->debug("Database ready");
+        return;
+    }
+
+    if (settings->value("run_without_secure_secrets_service").toBool()) {
+        for (auto &[name_, internal] : toLoad) {
+            auto name = secretName(name_, internal);
+            auto value = settings->value("secrets/" + name).toString();
+            if (value.isEmpty()) {
+                nhlog::db()->info("Restored empty secret '{}'.", name.toStdString());
+            } else {
+                std::unique_lock lock(secret_storage.mtx);
+                secret_storage.secrets[name.toStdString()] = value.toStdString();
+            }
+        }
+        QTimer::singleShot(0, [this] { loadSecrets({}); });
```

Comment:
Do you actually need the singleShots in all those places? If you do, could you add a comment everywhere?

Reply:
It seems so. Apparently signals are not being received unless I emit them from another slot.

Reply:
I added some comments.